### ISSUE TITLE
/signups index doesn't return data - breaking at an error message

### ISSUE THREAD

**Issue by chloealee** [DoSomethingArchive/legacy-website#6193]
#### Current Behavior

`/signups` index returns no data - breaking at `no reportback data found` and `no campaign data found` error messages. 

`https://thor.dosomething.org/api/v1/signups/23` returns 

```
{
  error: { 
    message: "No reportback data found."
  }
}
```

and `https://thor.dosomething.org/api/v1/signups/1` returns 

```
{
  error: {
    message: "No campaign data found."
  }
}
```
#### Desired Behavior

If the signup is not for a campaign, should not be included in the response. 
If there is no reportback, the reportback object should return `null` instead of the error message.

**Comment by chloealee** [comment]
@angaither @DFurnes I'm noticing that some reportbacks that are in the database are breaking with the error message `No reportback data found.` - could this be because of issue #6176 and why the index of `/signups` is breaking?

**Comment by DFurnes** [comment]
That's probably the root cause. I think a quick fix for now would be to see if you can catch that exception and return an empty array, rather than letting it bubble up to wherever that's reporting with that message.

/cc @weerd

**Comment by weerd** [comment]
@DFurnes @chloealee We had changed this to a [`try/catch` here](https://github.com/DoSomething/phoenix/pull/6168/files#diff-13b03397f87da0a2d3c8d9a98c71ae68R94) to address this issue earlier, but then in a later PR got changed to an [`if/else`](https://github.com/DoSomething/phoenix/pull/6191/files#diff-13b03397f87da0a2d3c8d9a98c71ae68R94)

**Comment by weerd** [comment]
Looks like you'll need a combo of first an `if/else` to check if you have an `rbid` to test and then the `try/catch` to see if it actually returns something with the one it tries.

**Comment by chloealee** [comment]
We changed it to an `if/else`[in this PR](https://github.com/DoSomething/phoenix/pull/6191) because the `try/catch` for some reason was sending back the incorrect reportback that wasn't associated with the correct signup. 

Would something like this be too messy to do? 

```
    if (isset($data->rbid) && is_numeric($data->rbid)) {
      try {
        $this->reportback = Reportback::get($data->rbid);
      } catch (Exception $e) {
        $this->reportback = null;
      }
    } else {
      $this->reportback = null;
    }
```

**Comment by weerd** [comment]
Not sure why the `try/catch` would have been breaking that though, since you're just passing it a specific `rbid`? That would likely be due to the query returning the wrong data regarding the reportback for the signup... all the `Reportback::get()` knows is the `id` of the item it's being asked to retrieve and if it was the wrong RB then that would be because the wrong `rbid` was passed... 

Unless I'm misunderstanding something :confused:

**Comment by DFurnes** [comment]
I don't think that's too messy, but we should plan to refactor those exceptions sometime soonish.

**Comment by DFurnes** [comment]
Could it have been casting a falsey value to `0` and trying to fetch that reportback ID perhaps?

**Comment by chloealee** [comment]
Yeah it was strange! It seemed to be just grabbing a random reportback and returning it - wasn't associated with the `sid` or the `uid`.

**Comment by angaither** [comment]
The way the reportback query is written if no rbid is passed it grabs all reportbacks 
So that was returning the first reportback for everyone who didn't reportback

**Comment by weerd** [comment]
oh, so we were passing a non-existent `rbid` to the `Reportback::get()`. Yeah, there should def be a logic check there then if no `rbid` exists to pass along.

The `try/catch` was necessary, because we had some faulty data in the DB where a Reportback existed but NO Reportback Item was associated with it and the system would fail, given that all Reportbacks _must_ have at least 1 Reportback Item.

**Comment by DFurnes** [comment]
> The way the reportback query is written if no rbid is passed it grabs all reportbacks 
> So that was returning the first reportback for everyone who didn't reportback

:boom: Ah ha!

**Comment by chloealee** [comment]
:mag: thanks for clarifying, team!

**Comment by weerd** [comment]
As an alternative to the not so pretty `if/else/try/catch` we could start by indicating that the `reportback` property is `null` and assign it something later if it fits the criteria... so:

```
$this->reportback = null;

if (isset($data->rbid) && is_numeric($data->rbid)) {
      try {
        $this->reportback = Reportback::get($data->rbid);
      } catch (Exception $e) {
        // report some error in watchdog or something...
      }
}
```

not sure if that's actually better, haha

**Comment by chloealee** [comment]
@weerd that does look better than what I have in the PR! But should I just clean it up/update once I fix #6176 ?
